src/main/java/com/aws/greengrass/mqttbridge/util/BatchedSubscriber.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.mqttbridge.util;
+
+import com.aws.greengrass.config.ChildChanged;
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.config.Subscriber;
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.config.WhatHappened;
+import lombok.NonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+
+/**
```

Comment:
I'm assuming there haven't been changes to this code compared to what is in Nucleus, so I am skipping review

Reply:
i did add one thing actually:  include a set of changed nodes in the callback.  use case is so we don't restart if ONLY the mqtt mapping changed:  https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/61/files/a94b383ee788dda0299fd4711c1833d91a18a786#diff-d90ee356c3fee8cf3ba84f0b41824f9ad008afef1d521de2e4e8769c87d4e0e3R191-R195.    @MikeDombo for awareness